### PR TITLE
integrated diya and abhis code

### PR DESCRIPTION
**Pull request by AbhinandG**
Please go over the code and approve the PR if it looks okay

### CODE REVIEW

**Comment by Akettle44** [comment]
Thanks for creating a PR! Here are some things I think could be addressed before merging:

- Clean the if/else if/ else structure in BertCustomHead and the training cell. If the training cell is converted to a function, we can have functionality (list / dictionary) that maps the desired task to the head and labels in both functions. Shoot me a message on discord if you want more info about this.

- Add cells proving that the loss decreases for each data loader / task. Bonus is that you hit the same metrics that you are expecting (maybe try using the huggingface version as a baseline?)

- @DiyadotSaha Would you be able to zip the cleaned data up and provide storage links so the rest of the team can download it? I think this is a good idea so that we don't have to run the cleaning code every time we want to train.

- Move all the imports to one place, right now certain libraries are imported multiple times. I tend to use the top of the notebook for this. This could be a subjective thing, I have seen other people do it on a per cell basis. IMO its messier per cell. 

- Address the question about the # of samples being different for the datasets from Monday's meeting.

At some point later this week, we need to have a discussion about the visualizations and how we can draw conclusions from them if the inputs are different. 

Thanks in advance,

Andrew

File: notebooks/data_setup.ipynb [hunk 1]
```diff
@@ -0,0 +1,655 @@
+{
```
Comment (Akettle44) [inline notebooks/data_setup.ipynb:1]: Format is raw here, see comments on PR main body for updates. Thanks.

**Review by Akettle44 — CHANGES_REQUESTED** [review]
Please address comments in main PR body.

**Comment by AbhinandG** [comment]
Most recent commit resolves issues 1, 2 and 4 raised by @Akettle44. Let me know if any more changes are to be made

**Comment by Akettle44** [comment]
> Most recent commit resolves issues 1, 2 and 4 raised by @Akettle44. Let me know if any more changes are to be made

Thanks for the update.  The bert head choice with the tasks looks much better now. I think we could do the same thing for the train function since we won't switch tasks in the middle of training. We can choose the task at the top of the train function. I am on mobile so I can't test yet, but it looks like the last cell errored out, do you know why that was? I don't see the model being explicitly tested anywhere to show that the data loaders and training process works.

**Pull request merged**